clusters.py

```diff
@@ -289,35 +293,58 @@ def ConvertToNumerical(sd: List[dataclasses.dataclass],
             d = np.nan
 
       data[i, j] = d
+
   return data
 
-def MakeDataClass(column_names: List,
-                  rows: List, features: List) -> List:
+def MakePandas(rows_of_data: List,
+               column_names: List[str],) -> pd.DataFrame:
   """
-  This function creates a new dataclass (StanfordData) object for each row of
-  data, and then returns a list of these objects.
+  Creates a Pandas DataFrame from a list of rows containing data.
 
   Args:
-    - column_names: A list of strings that represents the names of the columns
-      in the data.
-    - rows: A list of lists of strings that represents the rows of data.
-    - features: A subset of column_names that is needed for further processing
+      rows_of_data (List): A list of rows, where each row is an iterable containing data values.
+      column_names (List[str]): A list of column names for the DataFrame.
 
   Returns:
-    - A list of StanfordData objects, where each object represents a row of data,
-    and no column names are present.
+      pd.DataFrame: A Pandas DataFrame containing the converted data.
   """
-  column_names = ConvertSpaces(column_names) #To convert spaces in column names to _
-  StanfordData = dataclasses.make_dataclass('stanford_data', column_names)
-  all_stanford_data = [StanfordData(*rows[i]) for i in range(1, len(rows))]
+  
+  data = ConvertToNumerical_temp(rows_of_data, column_names)
+  data_df = pd.DataFrame(data, columns = column_names)
+
 
-  #To find the datapoints with apropriate age limit
-  age_data =  ConvertToNumerical(all_stanford_data, ['AgeAtTestDate'])
+  return data_df
 
-  good_age_rows,_ = np.where((age_data >=0) & (age_data <=100))
-  good_rows = [all_stanford_data[i] for i in good_age_rows]
-  data = ConvertToNumerical(good_rows, features)
+def RemoveRowsWithBadAges(df: pd.DataFrame) -> pd.DataFrame:
 
+  """
+  Removes rows from a DataFrame with invalid age values.
+
+  Args:
+      df (pd.DataFrame): The input DataFrame containing age data.
+
+  Returns:
+      pd.DataFrame: A new DataFrame with rows removed if their 'AgeAtTestDate' values are outside the valid range [0, 100].
+  """
+
+  df_good_age = df.drop(df.loc[(df['AgeAtTestDate'] < 0) | (df['AgeAtTestDate'] > 100)].index)
+  return df_good_age
+
+def RemoveRowsWithBCWorseAC(data: pd.DataFrame, threshold: int = 100) -> pd.DataFrame:
+  """
+  A function to drop the rows where Bone Conduction is greater than Air
+  Conduction by atleast 100dB
+
+  Args:
+  - data: A pandas dataframe on which HLossClassifier() function is run.
+  - threshold: The minimum difference between air conduction and bone conduction
+              loss.
+  Return:
+  - data: A pandas dataframe where the rows with bone conduction loss > air conduction
+  loss are dropped
+  """
+
+  data = data.drop(data.loc[data['R_PTA_BC_All'] - data['R_PTA_All'] >= threshold].index)
```

Comment:
Add a note that this only considers right-side loss.  Might be important to also do left.  But some other day.

It is probably worth adding a print statement telling of how many rows have been removed.  And the same for the function above.